### PR TITLE
Added Changes on SDK Side to register its Implemented Interfaces

### PR DESCRIPTION
**Pull request by varuntumbe**
Signed-off-by: varuntumbe1 <varuntumbe1@gmail.com>

Companion PR: https://github.com/opensearch-project/OpenSearch/pull/6191
### Description
1) Added a function getExtensionImplementedInterfaces() in src/main/java/org/opensearch/sdk/ExtensionsRunner.java
2) Modified InitializeExtensionResponse constructor to include implementedInterfaces.
3) Added a testcase testGetExtensionImplementedInterfaces.

### Issues Resolved
https://github.com/opensearch-project/opensearch-sdk-java/issues/319

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by varuntumbe** [comment]
Hi @dbwiddis ,

I have just pushed a fresh PR and I have added that extra code ( handling interfaces of super class ) as well. Tell me if we want to change anything.

Thanks and Regards
Varun

**Review by dbwiddis — APPROVED** [review]
LGTM!   Some notes:
1. For other maintainers: needs OpenSearch side PR merged first.
2. We realized that `BaseExtension` really shouldn't be implementing `ActionExtension`. That was incorrectly added in #434.   If you have time, feel free to remove that extra `implements` and update your test case in this PR to not include that (or alternately test the `HelloWorldExtension` which does implement that.).  But it's fine to leave this as-is and we'll handle that clean-up later.

**Comment by varuntumbe** [comment]
> LGTM! Some notes:
> 
> 1. For other maintainers: needs OpenSearch side PR merged first. (That is why gradle check is failing.)
> 2. We realized that `BaseExtension` really shouldn't be implementing `ActionExtension`. That was incorrectly added in #434.   If you have time, feel free to remove that extra `implements` and update your test case in this PR to not include that (or alternately test the `HelloWorldExtension` which does implement that.).  But it's fine to leave this as-is and we'll handle that clean-up later.

If its required, I can clean that up tomorrow.

Thanks and Regards
Varun

**Review by ryanbogan — DISMISSED** [review]
LGTM pending companion PR merge and gradle check re-run!  @varuntumbe thanks for the contribution!

**Comment by varuntumbe** [comment]
Thank you @ryanbogan and @dbwiddis. Need to retrigger build

**Comment by ryanbogan** [comment]
@varuntumbe I triggered the rebuild and there is still a failing compile error

**Comment by dbwiddis** [comment]
Yeah, usually GitHub is smart enough to compare PRs vs. latest main, but seems in the case of a file deletion it thinks you're trying to add that file back.   Rebasing to main (or deleting your local copy of TransportActions) should fix this.

**Comment by ryanbogan** [comment]
Looks like it's just missing javadocs now

**Comment by dbwiddis** [comment]
Hey @varuntumbe normally we would have waited for you to address the rebase (and the missing javadoc) but since we merged the OpenSearch companion PR it was breaking things for people who just updated.... so I went ahead and tidied up this PR.  Sorry to step on your toes, but the good news is this'll get merged shortly :)

**Review by dbwiddis — APPROVED** [review]

**Review by ryanbogan — APPROVED** [review]

**Comment by varuntumbe** [comment]
> Hey @varuntumbe normally we would have waited for you to address the rebase (and the missing javadoc) but since we merged the OpenSearch companion PR it was breaking things for people who just updated.... so I went ahead and tidied up this PR. Sorry to step on your toes, but the good news is this'll get merged shortly :)

Hi @dbwiddis,

I cant believe we missed that. Can we also make gradle precommit check like they have in opensearch ?

Thanks and Regards
Varun

**Comment by dbwiddis** [comment]
> I cant believe we missed that. 

Well, it was in the failing gradle check, see https://github.com/opensearch-project/opensearch-sdk-java/actions/runs/4359207215/jobs/7620732374

Our problem was not looking at the gradle check log and assuming that it was just the companion PR. :)

> Can we also make gradle precommit check like they have in opensearch ?

Our "check" only takes a few minutes so it's not really necessary.  Contributors should just do gradle check locally before they commit.

I'm really strongly leaning toward implementing checkstyle for javadocs (see comments on #496) and if so we could do a layered approach of first running spotless, then checkstyle, then gradle check.

**Comment by varuntumbe** [comment]
Hi @dbwiddis ,

Yes that makes more sense.

**Pull request merged**
